tool/release/notes/validate_release_notes.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+#
+# Copyright (C) 2022 Vaticle
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+git="git -C $BUILD_WORKSPACE_DIRECTORY"
+
+if $git diff --exit-code HEAD^ HEAD -- LATEST_RELEASE_NOTES.md; then
+    echo -n "Detected changes in the repository since the last time the release notes had been updated. "
+    echo "Please make sure those changes have been reflected in the LATEST_RELEASE_NOTES.md"
+    echo
+    echo "Latest change in LATEST_RELEASE_NOTES.md: $($git log -n 1 --pretty='%h (%aD)' -- LATEST_RELEASE_NOTES.md)"
+    echo "Latest commit: $($git log -n 1 --pretty='%h (%aD)')"
+    echo
+    echo "Aborting release."
+    exit 1
```

Comment:
Example message:

```
Detected changes in the repository since the last time the release notes had been updated. Please make sure those changes have been reflected in the LATEST_RELEASE_NOTES.md

Latest change in LATEST_RELEASE_NOTES.md: a1d14a1 (Fri, 5 May 2023 12:05:26 +0100)
Latest commit: 0919c8f (Fri, 5 May 2023 13:45:25 +0100)

Aborting release.
```